src/main/java/hudson/plugins/sonar/MsBuildSQRunnerInstallation.java

```diff
@@ -95,17 +99,42 @@ public String getDisplayName() {
     }
   }
 
-  public String getExecutable(Launcher launcher) throws IOException, InterruptedException {
+  public static String getScannerName() {
+    if (testExeName != null) {
+      return testExeName;
+    } else {
+      return SCANNER_EXE_NAME;
+    }
+  }
+
+  public String getToolPath(Launcher launcher) throws IOException, InterruptedException {
     return launcher.getChannel().call(new MasterToSlaveCallable<String, IOException>() {
       private static final long serialVersionUID = 1L;
 
       @Override
       public String call() throws IOException {
         String home = Util.replaceMacro(getHome(), EnvVars.masterEnvVars);
-        File exe = new File(home, exeName);
-        if (exe.exists()) {
-          return exe.getPath();
+
+        if (testExeName != null) {
+          File testScanner = new File(home, testExeName);
+          return testScanner.getPath();
+        }
+
+        File scannerNet46 = new File(home, SCANNER_EXE_NAME);
```

Comment:
Order between `SCANNER_EXE_NAME` and `SCANNER_DLL_NAME` is arbitrary and could be reversed because they are exclusives. `OLD_SCANNER_EXE_NAME` needs to be the last thought!